clients/cli/connections.c

```diff
@@ -2336,6 +2336,43 @@ typedef struct {
 	NMActiveConnection *active;
 } ActivateConnectionInfo;
 
+#if SYSTEMD_JOURNAL
```

Comment:
the logging backend is configurable at runtime. Maybe, NetworkManager should expose on D-Bus whether journal is used or not. For example via `NM_MANAGER_CAPABILITIES` properties.

Reply:
I don't think it is worth the complexity. This is a hint. If someone doesn't use journal on a platform with systemd-journald, they're on their own.

Reply:
Like rhel-7 (which compiles with journal support, but `logging.backend` defaults to `syslog`). So, this is hardly an unusual setup.

Reply:
Sigh. This is still rather odd setup. The logs end up in journal anyway, only stripped of useful metadata :( Eventually, everybody should be logging to journal.

In an effort to attempt complicating things to the point we'll have to ask the server, I've changed the condition to look at whether journal logging is the default.